test/factories/workflow-specific/PIM_WorkflowFactory_v1.t.sol

```diff
@@ -0,0 +1,565 @@
+        IPIM_WorkflowFactory_v1.PIMConfig memory pimConfig =
+            getDefaultPIMConfig();
+        pimConfig.withInitialLiquidity = true; // just to highlight what is being tested
+
+        uint preDeploymentCollateralBalance = token.balanceOf(address(this));
+        (IOrchestrator_v1 orchestrator, ERC20Issuance_v1 issuanceToken) =
+        factory.createPIMWorkflow(
+            workflowConfig,
+            paymentProcessorConfig,
+            logicModuleConfigs,
+            pimConfig
+        );
+        address fundingManager = address(orchestrator.fundingManager());
+        uint postDeploymentCollateralBalance = token.balanceOf(address(this));
+
+        // CHECK: curve HAS received initial collateral supply (and firstCollateralIn)
+        assertTrue(
+            preDeploymentCollateralBalance - postDeploymentCollateralBalance
+                == initialCollateralSupply + firstCollateralIn
+        );
+        // CHECK: recipient receives initialIssuanceSupply plus first purchase amount
+        assertGt(
+            issuanceToken.balanceOf(pimConfig.recipient),
+            bcProperties.initialIssuanceSupply
+        );
+        // CHECK: if recipient SELLS complete stack and BUYS BACK they get same amount of issuanceToken
+        vm.startPrank(pimConfig.recipient);
+        // get issuance balance before sale and rebuy
+        uint balanceBefore = issuanceToken.balanceOf(pimConfig.recipient);
+        // get static price before sale
+        uint staticPriceBefore = IBondingCurveBase_v1(fundingManager).getStaticPriceForBuying();
+        // get volume of first purchase in issuance token
+        uint firstPurchaseVolume = balanceBefore - pimConfig.bcProperties.initialIssuanceSupply;
+        issuanceToken.approve(fundingManager, firstPurchaseVolume);
+        uint collateralAmountOut = IRedeemingBondingCurveBase_v1(fundingManager)
+            .calculateSaleReturn(firstPurchaseVolume);
+        // use volume of first purchase to sell for collateral
+        IRedeemingBondingCurveBase_v1(fundingManager).sell(
+            firstPurchaseVolume, collateralAmountOut
+        );
+        // alice only has initial issuance supply left
+        assert(issuanceToken.balanceOf(pimConfig.recipient) == pimConfig.bcProperties.initialIssuanceSupply);
+        token.approve(fundingManager, collateralAmountOut);
+        uint issuanceAmountOut = IBondingCurveBase_v1(fundingManager)
+            .calculatePurchaseReturn(collateralAmountOut);
+        // now use the collateral from the sale to buy back
+        IBondingCurveBase_v1(fundingManager).buy(
+            collateralAmountOut, issuanceAmountOut
+        );
+        uint staticPriceAfter = IBondingCurveBase_v1(fundingManager).getStaticPriceForBuying();
+        // CHECK: if recipient SELLS complete stack and BUYS BACK they end up with same balance of issuanceToken
+        assertEq(
+            balanceBefore, issuanceToken.balanceOf(pimConfig.recipient)
+        );
```

Comment:
This assertion fails because there is a slight imprecision:
<img width="862" alt="image" src="https://github.com/user-attachments/assets/7f141703-12d2-46d5-8a02-9f69806e82bd">